internal/crypto/ptls/dialer.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package ptls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"net"
+	"time"
+)
+
+type Dialer interface {
+	IsReachableAndTLSValidationSucceeds(
+		address string,
+		certPool *x509.CertPool,
+		logger ErrorOnlyLogger,
+	) error
+}
+
+type ErrorOnlyLogger interface {
+	Error(msg string, err error, keysAndValues ...any)
+}
+
+type internalDialer struct {
+	dialer *net.Dialer
+}
+
+func NewDialer() *internalDialer {
+	return &internalDialer{
+		dialer: &net.Dialer{
+			Timeout: 15 * time.Second,
```

Comment:
It looks like we use a `30s` timeout in the `doExpensiveValidations` function of the jwtauthenticator and webhookauthenticator cache filler. Can we update this also to be 30s?